st2client/st2client/config_parser.py

```diff
@@ -138,6 +141,40 @@ def parse(self):
             # Config doesn't exist, return the default values
             return CONFIG_DEFAULT_VALUES
 
+        config_dir_path = os.path.dirname(self.config_file_path)
+
+        # Make sure the directory permissions == 0o770
+        if bool(os.stat(config_dir_path).st_mode & 0o777 ^ 0o770):
+            self.LOG.warn(
+                # TODO: Perfect place for an f-string
+                "The StackStorm configuration directory permissions are "
+                "insecure (too permissive)."
+                "\n\n"
+                "You can fix this by running:"
+                "\n\n"
+                "chmod 770 {config_dir}".format(config_dir=config_dir_path))
+
+        # Make sure the setgid bit is set
+        if not bool(os.stat(config_dir_path).st_mode & 0o2000):
+            self.LOG.warn(
+                # TODO: Perfect place for an f-string
+                "The StackStorm configuration directory permissions are "
+                "insecure (SGID bit not set)."
```

Comment:
I don't think it's a best to warn user about the absence of SGID bit.
It's a workaround that'll help creating file names with the same group under the config dir, but not really that much security-related.

Besides of that minor comment, looks good to me too :+1: